src/stores/Binding/Store.ts

```diff
@@ -945,22 +811,19 @@ const getInitialState = (
 
                 values.forEach(({ catalog_name, updated_at }) => {
                     const added =
-                        trackAddition ||
-                        (Object.keys(state.collectionMetadata).includes(
-                            catalog_name
-                        ) &&
-                            state.collectionMetadata[catalog_name].added);
-
-                    const triggered =
-                        backfilledCollections.includes(catalog_name) || added;
+                        addedCollections.includes(catalog_name) ||
+                        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                        state.collectionMetadata[catalog_name]?.added;
 
-                    updateCollectionMetadata(state, catalog_name, {
+                    state.collectionMetadata[catalog_name] = {
                         added,
                         sourceBackfillRecommended:
-                            triggered && isBeforeTrialInterval(updated_at),
+                            isBeforeTrialInterval(updated_at) &&
+                            (added ||
+                                backfilledCollections.includes(catalog_name)),
                         trialStorage: true,
```

Comment:
Can you throw in a comment why this is hardcoded to `true`? I _think_ I remember us talking about this but cannot recall.